runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -595,6 +595,12 @@ Table<K, V, M> openTable(@Nonnull final String namespace,
                              @Nullable final Class<M> mClass,
                              @Nonnull final TableOptions tableOptions)
             throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        Table<K, V, M> existingTable = (Table<K, V, M>) tableMap.get(fullyQualifiedTableName);
+        if (existingTable != null &&
+                existingTable.matchesParams(kClass, vClass, mClass, tableOptions)) {
+            return existingTable;
+        }
```

Comment:
Are there concurrency issues here if the multiple invocations to `openTable` overlap? It seems like we can still return two distinct Table objects if both invocations fail to observe the table in `tableMap`.

Reply:
Hi Zach, yes and that would be fine if in a rare scenario we end up returning two distinct objects (which is the default behavior currently). This is completely a performance optimization for the average serial table open case for most JVMs